promgen/forms.py

```diff
@@ -93,6 +93,57 @@ class Meta:
         }
 
 
+class ExporterLabelForm(ExporterForm):
+    class Meta:
+        model = models.ExporterLabel
+        exclude = ['exporter']
+
+        widgets = {
+            'name': forms.TextInput(attrs={'class': 'form-control', 'placeholder': 'Name'}),
+            'value': forms.TextInput(attrs={'class': 'form-control', 'placeholder': 'Value'}),
+            'is_parameter': forms.CheckboxInput()
+        }
+
+
+class BaseExporterLabelInlineFormSet(forms.BaseFormSet):
+    def clean(self):
+        """Checks that no two labels have the same name"""
+        if any(self.errors):
+            return
+        labels_names = {}
+        parameters_names = {}
+        for form in self.forms:
+            name = form.cleaned_data.get('name')
+            if not name:
+                continue
+            if form.cleaned_data.get('is_parameter'):
+                if name in parameters_names:
+                    e = ValidationError('Duplicated name', code='invalid')
```

Comment:
I think I need to try to test this on my own, but something seems a bit off at first glance. I *think* all the checks for duplicated can probably just be left to the database and you shouldn't need custom logic in here.